buildSrc/src/main/kotlin/io/spine/internal/gradle/test/Test.kt

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2021, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.internal.gradle.test
+
+import io.spine.internal.gradle.test.task.FastTest
+import io.spine.internal.gradle.test.task.SlowTest
+import org.gradle.api.tasks.TaskContainer
+import org.gradle.api.tasks.testing.Test
+import org.gradle.api.tasks.testing.TestDescriptor
+import org.gradle.api.tasks.testing.TestResult
+import org.gradle.api.tasks.testing.logging.TestExceptionFormat
+import org.gradle.kotlin.dsl.KotlinClosure2
+import org.gradle.kotlin.dsl.register
+
+/**
+ * Registers [slowTest][SlowTest] and [fastTest][FastTest] tasks in this [TaskContainer].
+ *
+ * Usage example:
+ *
+ * ```
+ * tasks {
+ *     registerTestTasks()
+ * }
+ * ```
+ */
+fun TaskContainer.registerTestTasks() {
+    register<FastTest>("fastTest").let {
+        register<SlowTest>("slowTest") {
+            shouldRunAfter(it)
+        }
+    }
+}
+
+/**
+ * Configures logging of this [Test] task.
+ *
+ * Enables logging of:
+ *  1. Standard `out` and `err` streams;
+ *  2. Thrown exceptions.
+ *
+ *  Additionally, after all the tests are executed, a short summary would be logged. The summary
+ *  consists of the number of tests and their results.
+ *
+ * Usage example:
+ *
+ *```
+ * tasks {
+ *     withType<Test> {
+ *         configureLogging()
+ *     }
+ * }
+ *```
+ */
+fun Test.configureLogging() {
```

Comment:
Let's reformat the whole PR into two files:

* `Logging.kt`, which would define the `configureLogging()` extension.
* `Tasks.kt`, which would contain all three routines: both of the implementations, and the `TaskContainer` extension. I assume that would allow us to reduce the visibility for `FastTest` and `SlowTest` types — for some reason, they are both `public` now, while we don't presume they should be used from outside.